app/lib/features/news/pages/add_news_page.dart

```diff
@@ -45,7 +45,8 @@ class AddNewsState extends ConsumerState<AddNewsPage> {
   //Build UI
   @override
   Widget build(BuildContext context) {
-    ref.listen(newsStateProvider, (prevState, nextState) async {
+    ref.listenManual(newsStateProvider, fireImmediately: true,
```

Comment:
why did you switch this to `listenManual`? `listen` has the same features and according to the [docs], listenManual isn't save to use here:
> As opposed to [listen](https://pub.dev/documentation/hooks_riverpod/latest/hooks_riverpod/WidgetRef/listen.html), [listenManual](https://pub.dev/documentation/hooks_riverpod/latest/hooks_riverpod/WidgetRef/listenManual.html) is not safe to use within the build method of a widget. Instead, [listenManual](https://pub.dev/documentation/hooks_riverpod/latest/hooks_riverpod/WidgetRef/listenManual.html) is designed to be used inside [State.initState](https://api.flutter.dev/flutter/widgets/State/initState.html) or other [State](https://api.flutter.dev/flutter/widgets/State-class.html) lifecycles.

If the plan was to move it to `initState` (which I think is a good goal), that move should also happen ;) .

Reply:
> why did you switch this to listenManual?

Because `listen` doesn't have `fireImmediately`
![Screenshot 2024-02-20 at 7 21 56 PM](https://github.com/acterglobal/a3/assets/51526480/19db0acf-9542-4eae-a525-4a834c71863b)

Reply:
> If the plan was to move it to initState (which I think is a good goal), that move should also happen ;) .

Ahh, Good notice. Let me move that to the `initState()` :)

Reply:
@kumarpalsinh25 [according to the docs listen does have that](https://pub.dev/documentation/hooks_riverpod/latest/hooks_riverpod/Ref/listen.html) ... That's how I knew about it, I didn't even know about `listenManual` before I saw it here ...

maybe we need to/want to upgrade our riverpod dependency?

Reply:
> @kumarpalsinh25 [according to the docs listen does have that](https://pub.dev/documentation/hooks_riverpod/latest/hooks_riverpod/Ref/listen.html) ... That's how I knew about it, I didn't even know about listenManual before I saw it here ...

👍️

> maybe we need to/want to upgrade our riverpod dependency?

Yes we can go with riverpod upgrade to get some benefits of latest changes in dependency.